src-tauri/src/document/intervals.rs

```diff
@@ -58,17 +58,28 @@ pub fn guess_control_interval_count(
         Some(next) => {
             let dx = next.x - this.x;
             let dy = next.y - this.y;
-            let dtheta = angle_modulus(next.heading - this.heading);
-            let heading_weight = 0.5; // arbitrary
+            let dtheta = angle_modulus(next.heading - this.heading).abs();
             let distance = dx.hypot(dy);
             let max_force = config.wheel_max_torque() / config.radius;
 
             // Default to robotConfig's max velocity and acceleration
             let mut max_vel = config.wheel_max_velocity() * config.radius;
             let mut max_accel = (max_force * 4.0) / config.mass; // times 4 for 4 modules
 
-            // Iterate through constraints to find applicable "Max Velocity" and "Max
-            // Acceleration" constraints
+            // find max wheel position radius for calculating max angular velocity
+            let max_wheel_position_radius = config
+                .modules
+                .iter()
+                .fold(0f64, |max, &module| max.max(module.x.hypot(module.y)));
+            let max_ang_vel = max_vel / max_wheel_position_radius;
+
+            // use peak angular velocity and dtheta to calculate the time
+            // for theta to travel a cubic hermite spline profile
+            // in which the max angular velocity is not exceeded as an estimate
+            let time = (1.5 * dtheta) / max_ang_vel;
```

Comment:
Where does this 1.5 number come from?

Reply:
It estimates the time needed to rotate `dtheta` without exceeding `max_ang_vel`. It comes from the cubic Hermite spline and moving around its variables.

/*
 * Proof for T = 1.5 * θ / ω:
 *
 * - The velocity function derived from the cubic Hermite spline is:
 *   v(t) = -6*θ*t^2 + 6*θ*t.
 *
 * - The peak velocity occurs at t = 0.5, where t∈[0, 1] :
 *   v(0.5) = 1.5*θ, which is the max angular velocity during the motion.
 * - To ensure this peak velocity does not exceed ω, we set:
 *   1.5 * θ = ω.
 * - The total time T needed to reach the final θ and
 *   not exceed ω is thus derived as:
 *   T = θ / (ω / 1.5) = 1.5 * θ / ω.
 *
 * This calculation ensures the peak velocity meets but does not exceed ω,
 * extending the time proportionally to meet this
 * requirement.
 */

Reply:
That proof should be included in the source code for context.